crates/config/src/host_component.rs

```diff
@@ -0,0 +1,46 @@
+use std::sync::Arc;
+
+use crate::{Error, Key, Path, Resolver, Result};
+
+mod wit {
+    wit_bindgen_wasmtime::export!("../../wit/ephemeral/spin-config.wit");
+}
+pub use wit::spin_config::add_to_linker;
+
+/// A component configuration interface implementation.
+pub struct ComponentConfig {
+    component_root: Path,
+    resolver: Arc<Resolver>,
+}
+
+impl ComponentConfig {
+    pub fn new(component_id: impl Into<String>, resolver: Arc<Resolver>) -> Result<Self> {
+        let component_root = Path::new(component_id)?;
+        Ok(Self {
+            component_root,
+            resolver,
+        })
+    }
+}
+
+impl<'a> wit::spin_config::SpinConfig for ComponentConfig {
+    fn get_config(
+        &mut self,
+        key: &str,
+    ) -> std::result::Result<std::string::String, wit::spin_config::Error> {
```

Comment:
nit: Would you mind adding `use` statements for standard library objects here? It makes reading things a bit easier.

Reply:
Whoops, this was vscode's generated trait impl for some reason... :thinking: